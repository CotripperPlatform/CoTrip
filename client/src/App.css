@import url("https://fonts.googleapis.com/css?family=Reem+Kufi&display=swap");
@import url("https://fonts.googleapis.com/css?family=Open+Sans:400,400i,600,700&display=swap");

:root {
  /* These are all the defaults for CoTripper. Below is an example of usage:
        background-color: var(--yellow);
    */

  /* Default color palette variables */
  --yellow: #ffcb06;
  --yellow-dark: #f2bc0f;
  --pink: #fb8370;
  --red: #d1495b;
  --purple: #6a4c93;
  --black: #252525;
  --white: #fafafa;
  --gray: #808080;
  --blue-lite: #3988ff;
  --gray-lite: #e5e5e5;
  --gray-dark: #c4c4c4;
  --green: #12db00;
  --white-hover: #fcf0ee;
  --orange-hover: #ff6a50;
  --purple-hover: #592d96;
  --red-hover: #d42e44;
  --yellow-hover: #f5c000;
  --pink-hover: #f86048;

  /* Background Graidients */
  --yellow-gradient: linear-gradient(#ffcb06 65%, #fff1c0);

  /* Default font */
  --default-font: "Reem Kufi", sans-serif;
  --paragraph-font: "Open Sans", sans-serif;
  /* Images */
  --comment-black: url("./assets/images/comment_black.png");
  --edit-button: url("./assets/images/edit_button.png");
  --global-nav-dm: url("./assets/images/global-nav_dm_icon.png");
  --global-nav-logo: url("./assets/images/global-nav_logo_icon.png");
  --global-nav-search: url("./assets/images/global-nav_search_icon.png");
  --heart-pink: url("./assets/images/heart_pink.png");
  --heart-white: url("./assets/images/heart_white.png");
  --heart-black: url("./assets/images/heart_black.png");
  --home-banner: url("./assets/images/home_banner.png");
  --profile-banner: url("./assets/images/profile_banner.png");
  --community-banner: url("./assets/images/community_banner.png");
  --login-logo: url("./assets/images/login_logo.png");
  --nav-arrow-purple: url("./assets/images/nav_arrow_purple.png");
  --nav-arrow-yellow: url("./assets/images/nav_arrow_yellow.png");
  --social-facebook: url("./assets/images/social_facebook.png");
  --social-instagram: url("./assets/images/social_instagram.png");
  --social-pinterest: url("./assets/images/social_pinterest.png");
  --splash-background: url("./assets/images/splash_background.png");
  --splash-background-mobile: url("./assets/images/splash_background_mobile.png");
  --text-box-search: url("./assets/images/text_box_search.png");
  /* Font Weights */
  --paragraph-weight: 400;
  --banner-weight: 700;
  --title-weight: 500;
}
