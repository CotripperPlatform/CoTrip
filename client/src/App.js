import React, { Component } from "react";
import { Route } from "react-router-dom";
import SplashPage from "./pages/SplashPage/SplashPage";
import ComingSoonPage from "./pages/ComingSoonPage/ComingSoonPage";
import "./App.css";
import { library } from "@fortawesome/fontawesome-svg-core";
import {
  faCheckSquare,
  faHeart,
  faTable,
  faComment,
  faCaretRight,
  faCaretLeft,
  faSearch,
  faAngleLeft,
  faAngleRight,
  faCommentDots
} from "@fortawesome/free-solid-svg-icons";
import { fab } from "@fortawesome/free-brands-svg-icons";
import { fas } from "@fortawesome/free-solid-svg-icons";
import { far } from "@fortawesome/free-regular-svg-icons";
library.add(
  fab,
  far,
  fas,
  faCheckSquare,
  faHeart,
  faTable,
  faComment,
  faCaretRight,
  faCaretLeft,
  faSearch,
  faAngleLeft,
  faAngleRight,
  faCommentDots
);

class App extends Component {
  render() {
    return (
      <div className="App">
        <main>
          <Route path="/" exact component={SplashPage}></Route>
          <Route path="/coming_soon" exact component={ComingSoonPage}></Route>
        </main>
      </div>
    );
  }
}

export default App;
