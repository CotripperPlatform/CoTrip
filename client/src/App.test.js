import React from "react";
import { shallow } from "enzyme";
import App from "./App";

const setUp = (props = {}) => {
  const component = shallow(<App />);
  return component;
};
describe("App component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setUp();
  });
  test("Is an Object", () => {
    const wrapper = setUp();
    expect(wrapper).toEqual(Object.prototype);
  });
  test("Snapshot renders", () => {
    const tree = setUp();
    expect(tree).toMatchSnapshot();
  });
});
