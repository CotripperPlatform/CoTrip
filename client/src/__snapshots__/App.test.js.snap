// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Snapshot renders 1`] = `
<div
  className="App"
>
  <main>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
  </main>
</div>
`;
